src/mat/4C_mat_inelastic_defgrad_factors.cpp

```diff
@@ -2878,164 +2850,147 @@ Core::LinAlg::Matrix<10, 1> Mat::InelasticDefgradTransvIsotropElastViscoplast::l
   // declare current right CG (tensor interpolated later on in each substep)
   Core::LinAlg::Matrix<3, 3> curr_CM(true);
 
-  // define current time step in the substepping procedure -> initialized to problem time step
-  double curr_dt = time_step_settings_.dt_;
-  // define time parameter of the substepping procedure
-  double t = 0;
-  // define count of times the substepping procedure has already halved the time step
-  int time_step_halving_counter = 0;
-  // define substep counter
-  unsigned int substep_counter = 1;
-  // define maximum number of substeps (gets adapted in the substepping scheme)
-  unsigned int max_num_of_substeps = 1;
+  // initialize substep parameters
+  SubstepParams substep_params = {
+      0,  // t = 0 (time parameter)
+      1,  // substep_counter = 1 (evaluation of first substep)
+      time_step_settings_
+          .dt_,  // curr_dt = time_step_settings_.dt_ (first substep length = full time step)
+      0,         // time_step_halving_counter = 0 (full time step, therefore no substep halving yet)
+      1,         // max_num_of_substeps = 1 (1 substep to evaluate: full time step)
+      0,         // iter = 0 (0 LNL iterations for the current substep)
```

Comment:
You can use designated initializers here: 

```suggestion
      .t = 0, 
     .substep_counter =  1, 
     ...
```